File: 874-backspace-string-compare/backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack_s, stack_t=[],[]
        for i in s:
            if (i=='#'):
                if stack_s:
                    stack_s.pop()
            else:
                stack_s.append(i)
        for j in t:
            if (j=='#'):
                if stack_t:
                    stack_t.pop()
            else:
                stack_t.append(j)
        if stack_s==stack_t:
            return True
        else:
            return False

File: 874-backspace-string-compare/backspace-string-compare.cpp
class Solution {
public:
    string match_string(string s) {
        stack<char> st;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '#') {
                if (!st.empty()) {
                    st.pop();
                }
            } else {
                st.push(s[i]);
            }
        }
        string resulting_str = "";
        while (!st.empty()) {
            resulting_str += st.top();
            st.pop();
        }
        reverse(resulting_str.begin(), resulting_str.end());
        return resulting_str;
    }

    bool backspaceCompare(string s, string t) {
        return match_string(s) == match_string(t);
    }
};